lsp-metals.el

```diff
@@ -192,21 +193,51 @@ Should be ignored if there is no open doctor window."
     (setq lsp-metals--current-buffer (current-buffer))
     (lsp-notify "metals/didFocusTextDocument" (lsp--buffer-uri))))
 
+(declare-function dap-debug (conf))
+(declare-function dap-register-debug-provider (name conf))
+
+(defun lsp-metals--debug-start (no-debug params)
+  (dap-register-debug-provider "scala" #'identity)
+  (-let [session-params (lsp-send-execute-command
+                         "debug-adapter-start"
+                         (gethash "arguments" params))]
+    (dap-debug
+     (list :debugServer (-> (gethash "uri" session-params)
+                            (split-string ":")
+                            cl-third
+                            string-to-number)
+           :type "scala"
+           :name (gethash "name" session-params)
+           :host "localhost"
+           :request "launch"
+           :noDebug no-debug))))
+
+(defun lsp-metals--model-refresh (_params)
+  (when (and (buffer-file-name)
```

Comment:
First argument for such handlers is a workspace, so no need to find it.

Reply:
I used it as a guard for the case in which the user has switched to a buffer not managed by metals. But now I think that I should change the code to refresh all buffers that are managed by metals - or at least this is what vscode seems to be doing.

Reply:
But we receive this notification for particular workspace, so we should refresh it only there, regardless of where the user is. Shouldn't we?

Reply:
Yes, I meant that we should refresh all buffers that are managed by the current metals workspace.